orangecontrib/spectroscopy/widgets/owspectra.py

```diff
@@ -1009,6 +1067,71 @@ def grid_apply(self):
         self.plot.showGrid(self.show_grid, self.show_grid, alpha=0.3)
         self.show_grid_a.setChecked(self.show_grid)
 
+    def find_peak_variables(self):
+        if self.maximum_point is None:
+            self.maximum_point = np.amax(self.data.X[:, self.data_xsind])
+        if self.minimum_point is None:
+            self.minimum_point = np.amin(self.data.X[:, self.data_xsind])
+        if self.start_point is None:
+            self.start_point = round(np.median(getx(self.data)), 2)
+
+    def peak_apply(self, position):
+        if self.viewtype == INDIVIDUAL:
+            Label_line = VerticalPeakLine()
+            Label_line.setMovable(True)
+            Label_line.setPen(pg.mkPen(color=QColor(Qt.black), width=2, style=Qt.DotLine))
+            Label_line.setSpan(mn=self.minimum_point, mx=self.maximum_point)
+            Label_line.label.setColor(color=QColor(Qt.black))
+            Label_line.label.setPosition(1)
+            Label_line.label.setMovable(True)
+            if position:
+                Label_line.setPos(position)
+                Label_line.label.setText(str(round(position, 3)))
+            else:
+                Label_line.setPos(self.start_point)
+                Label_line.label.setText(str(round(self.start_point, 3)))
+            self.plotview.addItem(Label_line)
+
+    def peak_apply_auto(self, prominence, minHeight, maxHeight, line_overlap):
+        if self.viewtype == INDIVIDUAL:
+            if line_overlap is None:
+                line_overlap = 0
+            x_axis = self.data_x
+            data = self.data.X[:, self.data_xsind]
+            peak = []
+            if np.shape(data) != (len(data),):
+                for i, val in enumerate(data):
```

Comment:
We should never go though the whole dataset, which can be VERY large, just through the curves shown.